src/daemon/database/actions.rs

```diff
@@ -487,7 +487,10 @@ pub fn db_update_presigned_tx(
 /// Insert a new Spend transaction in the database
 pub fn db_insert_spend(
     db_path: &PathBuf,
-    unvault_txs: &[&DbTransaction],
+    // FIXME: Rust newbie: i *don't* need to be moving this. So i want to take &[&T] but i can't
```

Comment:
You're not moving the transactions, are you?
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=1be844b0305ec94fe385c8efffb5fb74

Reply:
Hmm it does move as for example
```patch
diff --git a/src/daemon/database/actions.rs b/src/daemon/database/actions.rs
index 9b370dd..c1e436d 100644
--- a/src/daemon/database/actions.rs
+++ b/src/daemon/database/actions.rs
@@ -1055,7 +1055,7 @@ mod test {
             db_vault_by_unvault_txid(&db_path, &spend_tx_inputs[0].previous_output.txid)
                 .unwrap()
                 .unwrap();
-        db_insert_spend(&db_path, &[db_unvault.clone()], &spend_tx).unwrap();
+        db_insert_spend(&db_path, &[db_unvault], &spend_tx).unwrap();
         let spend_txid = spend_tx.inner_tx().global.unsigned_tx.txid();
         assert_eq!(
             db_list_spends(&db_path).unwrap().get(&spend_txid),
```
Does not compile (with a nice use-after-move error message)

Reply:
You were right, i'm actually moving it by creating an array each time! This compiles
```diff
diff --git a/src/daemon/database/actions.rs b/src/daemon/database/actions.rs
index 9b370dd..2c9a3da 100644
--- a/src/daemon/database/actions.rs
+++ b/src/daemon/database/actions.rs
@@ -1055,7 +1055,8 @@ mod test {
             db_vault_by_unvault_txid(&db_path, &spend_tx_inputs[0].previous_output.txid)
                 .unwrap()
                 .unwrap();
-        db_insert_spend(&db_path, &[db_unvault.clone()], &spend_tx).unwrap();
+        let arr = [db_unvault.clone()];
+        db_insert_spend(&db_path, &arr, &spend_tx).unwrap();
         let spend_txid = spend_tx.inner_tx().global.unsigned_tx.txid();
         assert_eq!(
             db_list_spends(&db_path).unwrap().get(&spend_txid),
@@ -1078,7 +1079,7 @@ mod test {
         // And this works with multiple unvaults too
 
         // Re-insert the previous one so we have many references to the first Unvault
-        db_insert_spend(&db_path, &[db_unvault.clone()], &spend_tx).unwrap();
+        db_insert_spend(&db_path, &arr, &spend_tx).unwrap();
```

Reply:
Removed the comment.